Plot min and max expected parameters

soumide1102 opened the pull request:
Currently the posterior plotting code allows users to plot one specific expected value for each parameter. However, sometimes the parameters are expected to lie in a specific range, instead of one single value. This PR adds two more options `--expected-parameters-min` and `--expected-parameters-max` which will plot the boundaries of the expected range using "dashdot" lines. 

An example plot is here  https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/test.png .

cdcapano commented:
@soumide1102 Could you explain the scenario where this would be needed? I don't quite follow.

soumide1102 commented:
@cdcapano For example, if you look at the BNS paper, many of the parameters have a range of possible values.. like mass1 is reported as 1.36-1.6 M_sun. So, if I want to plot the expected parameter in my plot with the BNS paper as reference, I would want to plot the expected range of values, because I can't pick a single value in such a case.

@micamu Adding a shading between the boundaries, sounds like a good idea. Thanks! :)

cdcapano commented:
@soumide1102 Ok... the thing that bothers me about that is the 2D box you get from the 1D marginal limits is not the same thing as the 2D marginal region. You can see that in your m1 vs m2 plot example. As a result, I don't really know how to compare the two: even if there is overlap between the 2D marginal posterior and the box, they may not actually be consistent. For example, the 1D marginal interval generally covers a narrower range of values than that spanned by the 2D marginal contour; so, if a 2D marginal extends outside of the box, does that mean you have inconsistent results?

I wouldn't be against doing this for the 1D marginal plots... just not sure about the 2D. Maybe this wouldn't be too bad if you shade, as Miriam suggests, and change the option names from "expected-range" to "shade-region". Anyway, try switching to shading first, then we can go from there.

soumide1102 commented:
@cdcapano @micamu I've made the expected parameters range a shaded region. I've done this for the 1D marginalized plots only. Doing it for the 2D looks confusing and complicated, so I left that as it was before.

An example plot with these changes is here https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/test3.png

micamu commented:
I think that looks a bit better... Maybe you could plot a dashed line in the 2D plots indicating the mean of the expected min-max, as a reference. But that is just an idea, you don't need to do that if you don't like it.
It looks fine for me, but let's give @cdcapano a chance to see the changes.

soumide1102 commented:
@cdcapano I've addressed your first two bullet points. The fc=(0.0, 0.0, 0.0, 0.5) part that you suggested did not work. The histograms were still solid after doing that. So I just adjusted the alpha value to be slightly transparent. Example plot with shaded region : https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/test4_zorder.png . With no shading https://sugwg-jobs.phy.syr.edu/~soumi.de/tmp/test4_noshade_zorder.png .
Regarding your third bullet : "shade_marginal" does not quite make it clear that it's an expected region for the parameter that we're plotting. So how about "expected-range-min", and "expected-range-max" for the two limits?

soumide1102 commented:
@cdcapano Did you get a chance to look at this? 
I don't know why landscape is failing. When I click on "Details" it gives me a server error.

cdcapano commented:
@soumide1102 Sorry for taking so long to get back to you about this. The new plots look good, but I'm still reluctant to add this to `plot_posterior`. Plot posterior already has a bunch of options, and I'm afraid this is going to muddle it more. I thought maybe changing the name of the option might make it clearer, but you're right it doesn't make it any cleared. Mostly, this seems like a feature that is useful for papers, but not necessarily for regular usage.

So, I suggest the following: can you add an executable who's sole purpose is to make 1D marginal plots? Have it call `create_marginalized_marginal` directly. In that program, include the extra arguments for expected-min, expected-max. Do not include the extra arguments in `create_multidim_plot`, and by extension `plot_posterior`/`plot_movie`.

tdent commented:
@soumide1102 @cdcapano can this be merged asap or closed?  Trying to reduce the number of open PRs.

soumide1102 commented:
Yes, closing this for now. Some of the changes might not also be compatible with the new pycbc inference code version.
